docs/spec/interfaces.md

```diff
@@ -0,0 +1,93 @@
+# Interface Contracts
+
+<!-- This image is sourced from https://drive.google.com/open?id=1Mgn0Oz85M9W1Pwfnw3u0OhZmIIxSxcM3lZUEZyPCoB0 -->
+![Interface Contracts Overview](images/interface-contracts-overview.svg)
+
+---
+
+## Subscribable
+
+A **Subscribable** resource will emit events that a _Subscription_ can direct
+to a _Targetable_ or _Sinkable_ resource.
+
+### Control Plane
+
+A **Subscribable** resource may be referenced in the _from_ section of a
+_Subscription_. The **Subscribable** resource MUST expose a
+_subscribable.channelable_ field (an ObjectReference) in its _status_ section.
+This _channelable_ field MUST be _Channelable_, and _channel_ MAY refer back to
+the **Subscribable** resource.
+
+### Data Plane
+
+A **Subscribable** resource is an event producer or forwarder, events that it
+produces or forwards are delivered via its _status.channel_ resource.
+
+---
+
+## Channelable
+
+A **Channelable** resource owns a list of subscribers for delivery of events.
+
+### Control Plane
+
+The **Channelable** resource has a list of _subscribers_ within the resources
+_spec.channelable_. In practice, the resolved _subscription_ _call_ and
+_result_ endpoints populate the Channelable's list of _subscribers_.
+
+### Data Plane
+
+**Channelable** resources will attempt delivery to each of they _subscribers_
+at least once, and retry if the callie returns errors.
+
+---
+
+## Targetable
+
+A **Targetable** resource represents a unit of work to be done on/to/with an
+event. They allow for event modification and flow control without taking
+ownership of the event. Typically a **Targetable** is a function.
+
+### Control Plane
+
+A **Targetable** resource MUST expose a _targetable.domainInternal_ field in
+its _status_ section. The _domainInternal_ value is an internal domain name
+that is capable of receiving event deliveries. **Targetable** resources may be
+referenced in the _call_ section of a _Subscription_.
+
+### Data Plane
+
+The **Targetable** resource does not ownership of the event that was passed to
+it, but it is allowed to modify the event by returning a response code of *2XX*
+and a mutated version back to the caller. The **Targetable** can prevent the
+event from continuing to the next stage by returning *2XX* and an empty body.
+All other response codes are considered an error and the caller will attempt to
+call again.
```

Comment:
We should probably define which errors indicate a soon retry vs a delayed retry. E.g. "404" or "403" probably won't go away any time soon, but "429" or "502" might be able to be retried shortly.

I'm not sure how we should treat 3xx, given that we'll be doing a POST (or equivalent).

Perhaps it's best to reference #502 here and not write too much on error types beyond:

```markdown
A successful (200) response will indicate that the event was successfully processed.
Non-2xx response codes indicate a failure, and event delivery may be retried (TODO:
see runtime contract).

In the event of a successful response with a body, the body will be interpreted as a CloudEvent response, and may be forwarded to the next stage by the delivery software.
```
Note that the delivery software may not always be a Channel, so it may make more sense to leave how to interpret the returned event somewhat abstract.